### PR TITLE
refactor: split up GET Bans into two separate routes.

### PR DESCRIPTION
**Pull request by JoshDevHub**
Refactor of the GET Bans method into two separate routes, one for leaderboard ID and one for user ID. Dropped route for GET all bans.

### CODE REVIEW

**Comment by TemperGrace** [comment]
When this is reviewed/merged, can we initialise a section in the wiki/documentation regarding Moderation, and document this functionality there? This is just one tool amongst what will become an entire toolset - will be important!

**Comment by RageCage64** [comment]
The docs for this don't change any documented functionality; the changes that this PR presents will be part of the generated API documentation in the comments.

File: LeaderboardBackend/Controllers/BansController.cs [hunk 1]
```diff
@@ -26,40 +26,46 @@ IBanService banService
...
-	/// <ul>
-	///   <li>passing <code>leaderboardId</code> will return all bans a Leaderboard has</li>
-	///   <li>passing <code>bannedUserId</code> will return all bans a User has</li>
-	/// </ul>
-	/// Don't specify both a Leaderboard ID and a User ID. Only specify one.
-	/// </remarks>
-	/// <param name="leaderboardId">Optional. Gets Bans a Leaderboard has.</param>
-	/// <param name="bannedUserId">Optional. Gets Bans a User has.</param>
+	/// <summary>Get bans by leaderboard ID</summary>
+	/// <param name="leaderboardId">Gets Bans a Leaderboard has.</param>
```
Comment (zysim) [inline LeaderboardBackend/Controllers/BansController.cs:30]:
```suggestion
	/// <param name="leaderboardId">The leaderboard ID.</param>
```

File: LeaderboardBackend/Controllers/BansController.cs [hunk 2]
```diff
@@ -26,40 +26,46 @@ IBanService banService
...
+							nameof(Conventions.Get))]
+	[HttpGet("leaderboard/{bannedUserId:Guid}")]
+	public async Task<ActionResult<List<Ban>>> GetBansByUser(Guid bannedUserId)
+	{
+		List<Ban> bans = await BanService.GetBansByUser(bannedUserId);
+
+		if (bans.Count == 0)
 		{
-			return Ok(await BanService.GetBans(bannedUserId));
+			return NotFound("No bans found for this User");
```
Comment (zysim) [inline LeaderboardBackend/Controllers/BansController.cs:65]:
```suggestion
			return NotFound("No bans found for this user");
```

**Review by zysim — CHANGES_REQUESTED** [review]
Just a bunch of style changes from me. I'd like for entities to always be referred to in lowercase, e.g. in docs and in response messages. (I should add this to the style guide).

**Review by zysim — APPROVED** [review]

**Comment by zysim** [comment]
> When this is reviewed/merged, can we initialise a section in the wiki/documentation regarding Moderation, and document this functionality there? This is just one tool amongst what will become an entire toolset - will be important!

While what Rage has said _is_ true, I do see a use for documenting our system for moderation. I could park a WIP page in the wiki for this.

**Pull request merged**